### PR TITLE
Remove tweak require symplify/smart-file-system

### PR DESCRIPTION
**Pull request by samsonasik**
@staabm this is remove `new SmartFileSystem` with new `Symfony\Component\Filesystem\Filesystem` for tweak of PR:

- https://github.com/rectorphp/rector-src/pull/3881

### CODE REVIEW

**Review by staabm — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -38,7 +38,6 @@
         "symfony/string": "^6.2.2",
         "symplify/easy-parallel": "^11.1",
         "symplify/rule-doc-generator-contracts": "^11.1",
-        "symplify/smart-file-system": "^11.2",
```
Comment (staabm) [inline composer.json:41]: do we need symfony/filesystem dependency now?

Comment (samsonasik) [reply composer.json:41]: symfony/filesystem is used in other part of code, so no need, also, already in vendor prefixed

https://github.com/rectorphp/rector/tree/569c7dc229ba5be538f24e39967a5aab3e2cb4c7/vendor/symfony/filesystem

**Comment by samsonasik** [comment]
@staabm resolved now 👍

**Comment by staabm** [comment]
maybe this is also the proper fix for the problem you worked arround in https://github.com/rectorphp/rector-src/pull/3865 ?
(and we may can drop the prefixed-build workaround now?)

File: src/Kernel/CachedContainerBuilder.php [hunk 2]
```diff
@@ -72,7 +72,7 @@ public function clearCache(): void
             return;
         }
 
-        $smartFileSystem = new SmartFileSystem();
-        $smartFileSystem->remove($cacheFiles);
+        $filesystem = new Filesystem();
+        $filesystem->remove($cacheFiles);
```
Comment (samsonasik) [inline src/Kernel/CachedContainerBuilder.php:76]: @staabm I am wondering why remove cache container called in normal `vendor/bin/rector` on as should only run on phpunit run?

**Comment by samsonasik** [comment]
I am not sure, as the remove() clear cache probably should not run as is limited on phpunit run only, and somehow it still by pass it.

**Pull request merged**